Lib/_collections_abc.py

```diff
@@ -498,6 +498,17 @@ def __getitem__(self, item):
             t_result = new_args[-1]
             t_args = new_args[:-1]
             new_args = (t_args, t_result)
+
+        # This happens in cases like `Callable[P, T][[P, str], bool][int]`,
+        # we need to flatten the result.
+        if len(new_args) > 2:
```

Comment:
We need to check that there is indeed a paramspec in `__parameters__`, otherwise this should just fall through and error.